lib/data/file/backend.js

```diff
@@ -65,11 +74,15 @@ export const backend = {
                 return callback(errors.InternalError);
             })
             .on('end', () => {
-                log.debug('finished writing data', { key });
-                fs.fsync(fd, () => {
-                    fs.close(fd);
-                    return callback(null, key);
-                });
+                streamEnd = true;
+                // Only in flowing mode when fs.write finished
+                if (request._readableState.flowing) {
```

Comment:
As @ThibaultRiviere and I discussed, we are not writing to a writeStream.  We could use fs.createWriteStream instead of just writing to the file and then we would have a stream to pipe to.